test/e2e/targetcleanup/target_cleanup_test.go

```diff
@@ -0,0 +1,50 @@
+package targetcleanuptest
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Testing cleaning up dir targets", func() {
+	When("A rebuild is done", func() {
+		It("removes and reload from the cache the dir target", func() {
+			useBobfile("with_dir_target")
+			defer releaseBobfile("with_dir_target")
+
+			b, err := BobSetup()
+			Expect(err).NotTo(HaveOccurred())
+
+			err = b.Build(ctx, "build")
```

Comment:
As the cache from the user system is used this might
not have a consistent behaviour.
Please setup the test to use a distinct buildinfo & artifact store.

Take a look how it's done here https://github.com/benchkram/bob/blob/d712b2fd0c9f1c2d064560d720567971670f25d0/test/e2e/artifacts/artifacts_suite_test.go#L65